test/unistd/access.c

```diff
@@ -65,14 +66,73 @@ int main() {
   printf("F_OK(%s): %d\n", "renamedfile", faccessat(AT_FDCWD, "renamedfile", F_OK, 0));
   printf("errno: %d\n", errno);
 
+  struct stat fileStats;
+  stat("fchmodtest", &fileStats);
+#if !defined(NODEFS) && !defined(NODERAWFS)
+  assert(fileStats.st_mode & 0666);
+#else
+  assert(fileStats.st_mode & 0644);
+#endif
+  
+  EM_ASM(
+    var fchmodstream = FS.open("fchmodtest", "r");
+#if WASMFS
+    FS.fchmod(fchmodstream, 0000);
+#else
+    FS.fchmod(fchmodstream.fd, 0000);
+#endif
+  );
+  stat("fchmodtest", &fileStats);
+  assert(!(fileStats.st_mode & 0777));
+
+#if !defined(NODEFS) && !defined(NODERAWFS)
```

Comment:
Again a comment on why the ifdef is present would help. Imagine someone reads this months or years from now - they might wonder if this works around a bug, or is an expected known limitation, or something else.